core/dbt/semver.py

```diff
@@ -427,6 +427,15 @@ def resolve_to_specific_version(requested_range, available_versions):
     return max_version_string
 
 
+def semver_regex_versioning(versions: List[str]) -> bool:
```

Comment:
Is this just for testing? Can we move it into the test file if that's the case?

Reply:
This is just for testing the various version types. I moved it into the testing file as that does make a lot more sense 😄